pkg/provider/kubernetes/client.go

```diff
@@ -120,17 +118,28 @@ func (c *Client) PushSecret(ctx context.Context, value []byte, typed v1.SecretTy
 			if typed != "" {
 				newType = typed
 			}
-			return c.createSecret(ctx, value, newType, remoteRef)
+			return c.createSecret(ctx, values, newType, remoteRef)
 		}
 		return getErr
 	}
+
+	if remoteRef.GetProperty() == "" {
+		// return gracefully if data is already in sync
+		if reflect.DeepEqual(extSecret.Data, values) {
+			return nil
+		}
+
+		// otherwise update remote property
+		return c.updateMap(ctx, extSecret, values)
+	}
+
 	// return gracefully if data is already in sync
-	if v, ok := extSecret.Data[remoteRef.GetProperty()]; ok && bytes.Equal(v, value) {
+	if v, ok := extSecret.Data[remoteRef.GetProperty()]; ok && bytes.Equal(v, values[remoteRef.GetProperty()]) {
 		return nil
 	}
 
 	// otherwise update remote property
-	return c.updateProperty(ctx, extSecret, remoteRef, value)
+	return c.updateProperty(ctx, extSecret, remoteRef, values[remoteRef.GetProperty()])
```

Comment:
here we lookup with `remoteRef.GetProperty()` which is not the `Match.secretKey` from above.